pkg/construct2/resource.go

```diff
@@ -3,6 +3,7 @@ package construct2
 type Resource struct {
 	ID         ResourceId
 	Properties Properties
+	Imported   bool
```

Comment:
Not blocking, but I don't think the construct package cares on the difference between an imported or otherwise immutable resource. Just naming though, can change later if we so desire.

Reply:
i didnt want to group them together without thinking through it a little bit more but its probably true that they will be handled the same way so i figure we can always change the naming